polyfill.test.js

```diff
@@ -119,3 +119,121 @@ tape("Array.prototype.with", (t) => {
     t.notDeepEqual(orig, copy);
     t.end();
 });
+
+tape("TypedArray.prototype.popped", (t) => {
+    const orig = new Int8Array([1, 2, 3]);
+    const expected = new Int8Array([1, 2]);
+
+    const copy = orig.popped();
+
+    t.deepEqual(copy, expected);
+    t.notEqual(orig, copy);
+    t.notDeepEqual(orig, copy);
+    t.end();
+});
+
+tape("TypedArray.prototype.pushed", (t) => {
+    const orig = new Int8Array([1, 2, 3]);

```

Comment:
Can we shift and use another TypedArray in each of these tests? (No need to test ehm all for each TypedArray unless you think about a clever way to do that, but I think it will make a test that currently looks very simple, complicated.

Reply:
Yeah, that could be a simple way to cover a variety of TypedArray types. Alternative I was thinking of was to wrap the tests in a loop like:

[Int8Array, UInt16Array, Float64Array].forEach(TypedArray => {
  tape(`${TypedArray.name}.prototype.pushed`, (t) => {
     const org = new TypedArray([1, 2, 3]);
     ...
  });
});

Reply:
I've updated the tests to cover more array types. Also added a two more cases for `.with` which actually found a bug in the polyfil, so fixed that too.